services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -1423,16 +1431,31 @@ class ApplicationServiceITest : DatabaseTest() {
         }
 
         @Test
-        fun `when application not in Allu should delete application`() {
+        fun `when application not in Allu should delete application and all its attachments`() {
             val hanke = hankeFactory.saveMinimal()
             val application =
                 applicationFactory.saveApplicationEntity(USERNAME, hanke = hanke, alluId = null)
+            // assert that application is saved to database
             assertThat(applicationRepository.findAll()).hasSize(1)
+            val attachment1 = attachmentFactory.saveAttachment(application.id!!)
+            val attachment2 = attachmentFactory.saveAttachment(application.id!!)
+            // assert that attachments are saved to cloud
+            assertTrue(applicationAttachmentContentService.exists(attachment1.blobLocation!!))
+            assertTrue(applicationAttachmentContentService.exists(attachment2.blobLocation!!))
+            // assert that attachment metadata is saved to database
+            assertThat(applicationAttachmentRepository.findByApplicationId(application.id!!))
+                .hasSize(2)
 
             applicationService.delete(application.id!!, USERNAME)
 
+            // assert that application is deleted from database
```

Comment:
I think the asserts themselves are pretty self-explanatory. Commenting on what the next line of code does is more distracting than useful.

Reply:
You're right. Removed comments. I added them for myself mostly to make things more clear in tests.